godtools/App/Share/Data/RealmDatabase/RealmDatabase.swift

```diff
@@ -15,18 +15,34 @@ class RealmDatabase {
     private let config: Realm.Configuration
     private let backgroundQueue: DispatchQueue = DispatchQueue(label: "realm.background_queue")
     
+    private lazy var isInMemory: Bool = {
+        return config.inMemoryIdentifier != nil
+    }()
+    private lazy var inMemRealm: Realm = {
```

Comment:
Hey Rachael,

I think I see why inMemRealm needs to be setup this way by exiting the ```background(async:``` method early.   Realm instances aren't thread safe and can only be accessed from the thread they're created from.  

The way inMemRealm is setup now is that it will be allocated on the thread it's first accessed from (since it's lazy) and then at that point forward it can only be accessed from that thread otherwise we will get the Realm accessed from incorrect thread crash.  I don't really want to constrain configurations with inMemory in this way.  I'd still like them to take advantage of creating Realm's on a background thread.

Instead, could you add some additional functionality to RealmDatabase.swift that would allow us to allocate a single threaded RealmDatabase instance from the thread it's created.  Then in the ```background(async:``` you can check for this and exit early with the Realm instance.  

This will allow for inMemory configurations to still take advantage of using the ```background(async``` and also give us the option to lock a RealmDatabase instance to a single thread from which it was allocated.